api_v3/lib/KalturaEntryService.php

```diff
@@ -148,6 +148,10 @@ protected function replaceResourceByEntry($dbEntry, $resource, $tempMediaEntry)
 		$dbEntry->setReplacementStatus(entryReplacementStatus::NOT_READY_AND_NOT_APPROVED);
 		if(!$partner->getEnabledService(PermissionName::FEATURE_ENTRY_REPLACEMENT_APPROVAL) || $dbEntry->getSourceType() == EntrySourceType::KALTURA_RECORDED_LIVE)
 			$dbEntry->setReplacementStatus(entryReplacementStatus::APPROVED_BUT_NOT_READY);
+		if ($dbEntry->getMediaType() == null && $tempDbEntry->getMediaType() != null)
```

Comment:
check if this is already happening on 
api_v3/services/BaseEntryService.php:399

Reply:
does not happen. this is relevent only when uploading in the usual way, where the token is already uploaded and the type (field inside KalturaBaseEntry) is automatic. In our case, we asked Atar to preselect KalturaMediaEntry so we don't go in that piece of code.